### PR TITLE
removed AsyncAtomicIndexedStore, added anti-entropy store and timesta…

### PR DESCRIPTION
**Pull request by cryogenian**
…mped store, made emptyStore a resource in indexed store spec, added indexed store hooks

The same as #4234, but with one commit only 

[ch8215]

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -280,6 +280,7 @@ lazy val core = project
 /** Implementations of the Quasar API. */
 lazy val impl = project
   .settings(name := "quasar-impl")
+  .settings(parallelExecution in Test := false)
```
Comment (wemrysi) [inline build.sbt]: Would this be necessary if we used distinct ports for the nodes in each of the suites that use `atomix`?

Comment (cryogenian) [reply build.sbt]: This must be _partially hardcoded_, i.e. `AtomixClusterSpec` uses `6000 + x` and `AntiEntropyStoreSpec` uses `7000 + x`

**Comment by cryogenian** [comment]
@wemrysi Made ports in `AtomixSpec` and `AntiEntropyStore` being per-module unique for every node, removed `parallelExecution in Test := false`

**Comment by cryogenian** [comment]
@wemrysi 

+ `random` takes two element set instead of full `peers` (it might be one element set, but this isn't a big deal I think) 
+ `use` instead of `allocated` in tests
+ Changed log message
+ Changed enqueue function a bit

**Comment by cryogenian** [comment]
@wemrysi 
+ Extracted config into distinct module
+ Added `NodeAddress` and `NodeInfo.address: NodeAddress`, since we don't need ids for seeds

File: impl/src/main/scala/quasar/impl/cluster/Atomix.scala [hunk 2]
```diff
@@ -92,8 +96,12 @@ object Atomix extends Logging {
     def peers: F[Set[MemberId]] =
       F.delay(service.getMembers.asScala.to[Set].map(_.id))
 
-    def random: F[Set[MemberId]] =
-      peers
+    def random: F[Set[MemberId]] = for {
+      ps <- peers
```
Comment (wemrysi) [inline impl/src/main/scala/quasar/impl/cluster/Atomix.scala]: ⛳️ I think it'd be a bit more natural to convert `ps` to a `Vector` and then project indices out of it below.

File: impl/src/main/scala/quasar/impl/cluster/Atomix.scala [hunk 3]
```diff
@@ -140,24 +148,32 @@ object Atomix extends Logging {
...
-            case None => F.delay {
-              log.warn(s"incorrect message received\neventType ::: ${eventType}\nsource ::: ${addr}\nbits ::: ${a.toHex}")
-            }
-            case Some(_) if size >= maxItems => F.delay {
-              log.warn(s"there are too much events in subscription queue\neventType ::: ${eventType}\nsize ::: ${size}")
-            }
-            case Some(id) => q.enqueue1((id, d.value))
+      def decodeBits(a: BitVector, cont: P => F[Unit]): F[Unit] = Codec[P].decode(a) match {
+        case Attempt.Failure(_) =>
+          F.delay(log.warn(s"malformed payload received\neventType ::: ${eventType}\nbits ::: ${a.toHex}"))
```
Comment (wemrysi) [inline impl/src/main/scala/quasar/impl/cluster/Atomix.scala]: We don't need the newlines here (and elsewhere). Will also make parsing the logs a bit easier without them.

**Pull request merged**